sparkles/yoshi.py

```diff
@@ -0,0 +1,220 @@
+    aimpoints = {'ACIS-I': (970.0, 975.0, 3),
+                 'ACIS-S': (210.0, 520.0, 7),
+                 'HRC-S': (2195.0, 8915.0, 2),
+                 'HRC-I': (7590.0, 7745.0, 0)}
+    chip_x, chip_y, chip_id = aimpoints[ocat['instr']]
+    targ.update({'chipx': chip_x, 'chipy': chip_y, 'chip_id': chip_id})
+
+    # Nominal roll is not quite the same targ and aca but don't care.
+    targ['roll_targ'] = nominal_roll(ocat['ra'],
+                                     ocat['dec'],
+                                     obs_date)
+
+    # Set dither from defaults and override if defined in OCAT
+    if ocat['instr'].startswith('ACIS'):
+        targ.update({'dither_y': 8, 'dither_z': 8})
+    else:
+        targ.update({'dither_y': 20, 'dither_z': 20})
+
+    if ocat['dither'] == 'Y':
+        targ.update({'dither_y': ocat['y_amp'] * 3600,
+                    'dither_z': ocat['z_amp'] * 3600})
+    if ocat['dither'] == 'N':
+        targ.update({'dither_y': 0, 'dither_z': 0})
+
+    return targ
+
+
+def run_one_yoshi(
+    *, obsid,
+    detector, chipx, chipy, chip_id,
+    ra_targ, dec_targ, roll_targ,
+    offset_y, offset_z, sim_offset, focus_offset,
+    dither_y, dither_z,
+    obs_date, t_ccd, man_angle,
+    **kwargs
+):
+    """
+    Run proseco and sparkles for an observation request in a roll/temperature/man_angle
+    scenario.
+    :param obsid: obsid
+    :param detector: detector (ACIS-I|ACIS-S|HRC-I|HRC-S)
+    :param chipx: chipx from zero-offset aimpoint table entry for obsid
+    :param chipy: chipy from zero-offset aimpoint table entry for obsid
+    :param chip_id: chip_id from zero-offset aimpoint table entry for obsid
+    :param ra_targ: target RA (degrees)
+    :param dec_targ: target Dec (degrees)
+    :param roll_targ: target Roll (degrees)
+    :param offset_y: target offset_y (arcmin)
+    :param offset_z: target offset_z (arcmin)
+    :param sim_offset: SIM Z offset (steps)
+    :param focus_offset: SIM focus offset (steps)
+    :param dither_y: Y amplitude dither (arcsec)
+    :param dither_z: Z amplitude dither (arcsec)
+    :param obs_date: observation date (for proper motion and ACA offset projection)
+    :param t_ccd: ACA CCD temperature (degrees C)
+    :param man_angle: maneuver angle (degrees)
+    :param **kwargs: additional keyword args to update or override params from
+        yoshi for call to get_aca_catalog()
+    :returns: dictionary of (ra_aca, dec_aca, roll_aca,
+                             n_critical, n_warning, n_caution, n_info,
+                             P2, guide_count)
+    """
+    from proseco import get_aca_catalog
+
+    params = convert_yoshi_to_proseco_params(
+        obsid, detector,
+        chipx, chipy, chip_id,
+        ra_targ, dec_targ, roll_targ,
+        offset_y, offset_z, sim_offset, focus_offset,
+        dither_y, dither_z,
+        obs_date, t_ccd, man_angle
+    )
+
+    # Update or override params from yoshi for call to get_aca_catalog
```

Comment:
Since your preference is to get the params and mine was to get the aca/acar, I would figure the convention for getting aca would just end up being:
```
yp = get_yoshi_params_from_ocat(obsid=obsid)
pp = convert_yoshi_to_proseco_params(obsid=obsid, t_ccd=t_ccd, man_angle=man_angle, **yp)
aca = get_aca_catalog(**pp)
```
Not sure where to add that in the docs...  Also, does convert_yoshi_to_proseco_params deserve its own test?

Reply:
That's the idiom. I'm out of steam to document that, but I think the flow is pretty obvious and this is basically for internal use.

`convert_yoshi_to_proseco_params` is now tested.